pkg/commands/copy_test.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package commands
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/GoogleContainerTools/kaniko/pkg/dockerfile"
+	"github.com/GoogleContainerTools/kaniko/testutil"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/moby/buildkit/frontend/dockerfile/instructions"
+	"github.com/sirupsen/logrus"
+)
+
+var copyTests = []struct {
+	name           string
+	SourcesAndDest []string
+	expectedDest   []string
+}{
+	{
+		name:           "copy foo into tempCopyExecuteTest/",
+		SourcesAndDest: []string{"foo", "tempCopyExecuteTest/"},
+		expectedDest:   []string{"foo"},
```

Comment:
i am guessing you are going to fix this later and just adding tests for now?
the expectedDest should be `temp/CopyExecuteTest/foo` right?

We should also add a test case where we are copying a dir to dir.
1. COPY dir dir/  - shd copy dir into dir/
2. COPY dir2 dir - shd copy dir2 as dir

It will easy to create these file layouts with a test util.

Reply:
Actually no, in this case the expectedDest is correct.  And yes, we should add some more tests later to make it clearer.

The situation is that:
copy thing DirThatExists/
result is DirThatExists/thing

copy thing DirThatExists
result is DirThatExists/thing

copy thing DirThatDoesNOTExist/
result is DirThatDoesNOTExist/thing

copy thing DirThatDoesNOTExist
result is thing gets renamed to DirThatDoesNOTExcist